Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/FormWithRedirect/Ideal/BanksComponent.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  IdealComponent.swift
+//  PrimerSDK
+//
+//  Created by Alexandra Lovin on 06.11.2023.
+//
+
+import Foundation
+
+public enum BanksCollectableData: FormCollectableData {
+    case bankId(bankId: String)
+    case bankFilterText(text: String)
+}
+
+public class BanksComponent: PrimerHeadlessFormComponent {
+
+    public typealias T = BanksCollectableData
+    let paymentMethodType: PrimerPaymentMethodType
+    
+    public weak var errorDelegate: PrimerHeadlessErrorableDelegate?
+    public weak var validationDelegate: PrimerHeadlessValidatableDelegate?
+    public weak var stepDelegate: PrimerHeadlessSteppableDelegate?
+
+    public private(set) var nextDataStep: BanksStep = .loading
+    private(set) var banks: [IssuingBank] = []
+    private(set) var bankId: String?
+    private let createWebRedirectComponent: () -> WebRedirectComponent
```

Comment:
In this moment, it seems this does not need to be injected/lazy? Do we see a future where this will take some params at init time?
Related to an earlier comment regarding the Web Component implementation?

Reply:
Yes, the WebComponent implementation and handling is still wip and final implementation will followup in another PR later this week.
As for the lazy init, the createWebRedirectComponent only get called when the bank get selected and the tokenisation has been successfully performed.